src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1210,6 +1219,9 @@ internal TextLoader(IHostEnvironment env, Options options = null, IMultiStreamSo
                 }
             }
 
+            if (options.DecimalMarker == ',' && _separators.Contains<char>(','))
+                throw _host.ExceptUserArg(nameof(Options.DecimalMarker), "Decimal marker and separator cannot be the same '{0}' character.", options.DecimalMarker);
```

Comment:
Perhaps it's better to check _separators.Contains(options.DecimalMarker)? Specially if you plan to allow other decimal markers... and also because I'm not sure if we could have "." as separator... so checking it the way I suggest would be more general.

Reply:
The '.' character is not listed as an illegal here, checking `_separators.Contains(options.DecimalMarker)` is a good point.
https://github.com/dotnet/machinelearning/blob/a8a9b54b0a05708f988db5366a20c11114fe410e/src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs#L1402